Library/Homebrew/os.rb

```diff
@@ -13,6 +13,32 @@ def self.linux?
     RbConfig::CONFIG["host_os"].include? "linux"
   end
 
+  def self.family
+    return :mac if mac?
+    return :linux if linux?
+
+    :generic
+  end
+
+  def self.version
+    return Mac.full_version if mac?
+    return Linux.os_version if linux?
+
+    nil
+  end
+
+  def self.build_system_info
+    build_system_info = {
+      "os"         => OS.family,
+      "os_version" => OS.version&.to_s,
+    }
+
+    build_system_info.merge! Mac.build_system_info if mac?
+    build_system_info.merge! Linux.build_system_info if linux?
```

Comment:
Would rather push this logic into the `extend/os` pattern we've been using already rather than more `if/`else` in this file.